games-fps/yamagi-quake2/yamagi-quake2-7.10.ebuild

```diff
@@ -0,0 +1,134 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit desktop eutils
+
+CTF_V=1.05
+ROGUE_V=2.04
+XATRIX_V=2.05
+
+DESCRIPTION="Quake 2 engine focused on single player"
+HOMEPAGE="https://www.yamagi.org/quake2/"
+SRC_URI="https://deponie.yamagi.org/quake2/quake2-${PV}.tar.xz
+	ctf? ( https://deponie.yamagi.org/quake2/quake2-ctf-${CTF_V}.tar.xz )
+	rogue? ( https://deponie.yamagi.org/quake2/quake2-rogue-${ROGUE_V}.tar.xz )
+	xatrix? ( https://deponie.yamagi.org/quake2/quake2-xatrix-${XATRIX_V}.tar.xz )"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+client ctf dedicated ogg openal rogue xatrix"
+REQUIRED_USE="|| ( client dedicated )"
+
+RDEPEND="sys-libs/zlib:0=
+	client? (
+		media-libs/libsdl2[opengl,video]
+		virtual/opengl
+		ogg? (
+			media-libs/libogg
+			media-libs/libvorbis
+		)
+		openal? ( media-libs/openal )
+		!openal? ( media-libs/libsdl2[sound] )
+	)
+"
+
+DEPEND="${RDEPEND}"
+
+S="${WORKDIR}/quake2-${PV}"
+
+PATCHES=( "${FILESDIR}"/${PN}-respect-flags.patch )
+DOCS=( CHANGELOG CONTRIBUTE README.md )
+
+mymake() {
+	emake \
+		VERBOSE=1 \
+		DLOPEN_OPENAL=no \
+		WITH_CDA=no \
+		WITH_SYSTEMWIDE=yes \
+		WITH_SYSTEMDIR="${EPREFIX}"/usr/share/games/quake2 \
+		WITH_ZIP=yes \
+		WITH_OGG=$(usex ogg) \
+		WITH_OPENAL=$(usex openal) \
+		"$@"
+}
+
+src_prepare() {
+	local addon
+	for addon in ctf rogue xatrix; do
+		use ${addon} || continue
+		pushd "${WORKDIR}"/quake2-${addon}-* >/dev/null || die
+		eapply -l -- "${FILESDIR}"/${PN}-addon-respect-flags.patch
+		popd >/dev/null || die
+	done
+
+	default
+}
+
+src_compile() {
+	local targets="game"
+	use client && targets+=" client ref_gl1 ref_gl3"
+	use dedicated && targets+=" server"
+
+	mymake config
+	mymake ${targets}
+
+	local addon
+	for addon in ctf rogue xatrix; do
+		use ${addon} || continue
+		pushd "${WORKDIR}"/quake2-${addon}-* >/dev/null || die
+		emake VERBOSE=1
```

Comment:
`emake -C ...` instead of pushd/popd.

Afterwards, you may leverage using three inline `use ... && emake -C ...` instead of the loop (your choice).